PyDynamic/uncertainty/propagate_convolution.py

```diff
@@ -0,0 +1,147 @@
+# -*- coding: utf-8 -*-
+
+import numpy as np
+
+from .propagate_filter import _fir_filter
+
+__all__ = ["convolve_unc"]
+
+
+def convolve_unc(x1, U1, x2, U2, mode="full"):
+    """
+    An implementation of the discrete convolution of two signals with uncertainty propagation.
+    The code builds on PyDynamic.uncertainty._fir_filter, as convolution and filter application
```

Comment:
If we would move `_fir_filter` to a public function, we would then have three functions (`fir_filter`, `fir_filter_diag`, `FIRuncFilter`. ) with very similar scope - which could to lead to confusion on the user-side. 
But the benefit of this would be, that we can then link this `fir_filter`-method in the corresponding documentations (``FIRuncFilter` and `convolve_unc`).

Reply:
The main question then would be, does the mentioning of `fir_filter` provide new insights or is that more of a technical detail? Or maybe even more: would a user in some cases rather call `fir_filter` directly?

Reply:
So we adapted the docstring to keep it one abstraction layer higher and keep `_fir_filter` private.